stdlib/importlib/metadata/__init__.pyi

```diff
@@ -109,8 +134,14 @@ if sys.version_info >= (3, 8):
     def distributions(
         *, context: None = ..., name: str | None = ..., path: list[str] = ..., **kwargs: Any
     ) -> Iterable[Distribution]: ...
-    def metadata(distribution_name: str) -> Message: ...
+
+    if sys.version_info >= (3, 10):
+        def metadata(distribution_name: str) -> PackageMetadata: ...
+        def entry_points() -> SelectableGroups: ...
```

Comment:
Like `SelectableGroups.entry_points`, this takes `**params` and can return either an `EntryPoints` (if there are any params) or `SelectableGroups` (if there aren't).

Reply:
Added an appropriate `@overload`.